pointcloud/basic_octree.go

```diff
@@ -182,3 +189,158 @@ func (octree *BasicOctree) CollidesWithGeometry(geom spatialmath.Geometry, thres
 	}
 	return false, errors.New("unknown octree node type")
 }
+
+// Pose returns the pose of the octree.
+func (octree *BasicOctree) Pose() spatialmath.Pose {
+	return octree.center
+}
+
+// AlmostEqual compares the octree with another geometry and checks if they are equivalent.
+func (octree *BasicOctree) AlmostEqual(geom spatialmath.Geometry) bool {
+	// TODO
+	return false
+}
+
+// Transform recursively steps through the octree and transforms it by the given pose.
+func (octree *BasicOctree) Transform(p spatialmath.Pose) spatialmath.Geometry {
+	var transformedOctree *BasicOctree
+	switch octree.node.nodeType {
+	case internalNode:
+		transformedOctree = &BasicOctree{
+			basicOctreeNode{
```

Comment:
Nodes should only be created through the `newInternalNode`, `newLeafNodeEmpty` and `newLeafNodeFilled` functions found in `basic_octree_utils.go`

Reply:
See `splitIntoOctants` for an example.